diff -up nss/cmd/certutil/certutil.c.keep_1024_default nss/cmd/certutil/certutil.c
--- nss/cmd/certutil/certutil.c.keep_1024_default	2016-08-15 18:17:08.644962339 +0200
+++ nss/cmd/certutil/certutil.c	2016-08-15 18:18:27.007720980 +0200
@@ -40,7 +40,7 @@
 #define MIN_KEY_BITS 512
 /* MAX_KEY_BITS should agree with MAX_RSA_MODULUS in freebl */
 #define MAX_KEY_BITS 8192
-#define DEFAULT_KEY_BITS 2048
+#define DEFAULT_KEY_BITS 1024
 
 #define GEN_BREAK(e) \
     rv = e;          \
