diff -up ./nss/cmd/certutil/certutil.c.keep_1024_default ./nss/cmd/certutil/certutil.c
--- ./nss/cmd/certutil/certutil.c.keep_1024_default	2019-08-27 16:55:53.649354534 -0700
+++ ./nss/cmd/certutil/certutil.c	2019-08-28 09:55:35.512141648 -0700
@@ -42,7 +42,7 @@
 #define MIN_KEY_BITS 512
 /* MAX_KEY_BITS should agree with RSA_MAX_MODULUS_BITS in freebl */
 #define MAX_KEY_BITS 8192
-#define DEFAULT_KEY_BITS 2048
+#define DEFAULT_KEY_BITS 1024
 
 #define GEN_BREAK(e) \
     rv = e;          \
