diff -up nss/lib/ssl/sslsock.c.disable-ems nss/lib/ssl/sslsock.c
--- nss/lib/ssl/sslsock.c.disable-ems	2018-02-28 12:38:42.137098152 +0100
+++ nss/lib/ssl/sslsock.c	2018-02-28 12:39:48.284614954 +0100
@@ -1231,7 +1231,10 @@ SSL_OptionSetDefault(PRInt32 which, PRIn
             break;
 
         case SSL_ENABLE_EXTENDED_MASTER_SECRET:
+#if 0
+/* No-Op until we have a compatible softokn. */
             ssl_defaults.enableExtendedMS = val;
+#endif
             break;
 
         case SSL_ENABLE_SIGNED_CERT_TIMESTAMPS:
