diff -up nss/lib/ssl/sslsock.c.disable-ems nss/lib/ssl/sslsock.c
--- nss/lib/ssl/sslsock.c.disable-ems	2016-10-10 16:38:11.826109468 +0200
+++ nss/lib/ssl/sslsock.c	2016-10-10 16:43:06.859566935 +0200
@@ -82,6 +82,7 @@ static sslOptions ssl_defaults = {
     PR_TRUE,               /* reuseServerECDHEKey */
     PR_FALSE,              /* enableFallbackSCSV */
     PR_TRUE,               /* enableServerDhe */
+/* Keep extended-master-secret disabled until we have a compatible softokn. */
     PR_FALSE,              /* enableExtendedMS    */
     PR_FALSE,              /* enableSignedCertTimestamps */
     PR_FALSE,              /* requireDHENamedGroups */
@@ -827,7 +828,10 @@ SSL_OptionSet(PRFileDesc *fd, PRInt32 wh
             break;
 
         case SSL_ENABLE_EXTENDED_MASTER_SECRET:
+#if 0
+/* No-Op until we have a compatible softokn. */
             ss->opt.enableExtendedMS = on;
+#endif
             break;
 
         case SSL_ENABLE_SIGNED_CERT_TIMESTAMPS:
@@ -1276,7 +1280,10 @@ SSL_OptionSetDefault(PRInt32 which, PRBo
             break;
 
         case SSL_ENABLE_EXTENDED_MASTER_SECRET:
+#if 0
+/* No-Op until we have a compatible softokn. */
             ssl_defaults.enableExtendedMS = on;
+#endif
             break;
 
         case SSL_ENABLE_SIGNED_CERT_TIMESTAMPS:
