diff -up nss/gtests/pk11_gtest/manifest.mn.disable-curve25519-gtests nss/gtests/pk11_gtest/manifest.mn
--- nss/gtests/pk11_gtest/manifest.mn.disable-curve25519-gtests	2018-02-28 12:41:59.036683147 +0100
+++ nss/gtests/pk11_gtest/manifest.mn	2018-02-28 12:42:58.059359706 +0100
@@ -9,7 +9,6 @@ MODULE = nss
 CPPSRCS = \
       pk11_aeskeywrap_unittest.cc \
       pk11_chacha20poly1305_unittest.cc \
-      pk11_curve25519_unittest.cc \
       pk11_ecdsa_unittest.cc \
       pk11_encrypt_derive_unittest.cc \
       pk11_export_unittest.cc \
diff -up nss/gtests/ssl_gtest/ssl_ecdh_unittest.cc.disable-curve25519-gtests nss/gtests/ssl_gtest/ssl_ecdh_unittest.cc
--- nss/gtests/ssl_gtest/ssl_ecdh_unittest.cc.disable-curve25519-gtests	2018-02-28 12:41:59.036683147 +0100
+++ nss/gtests/ssl_gtest/ssl_ecdh_unittest.cc	2018-02-28 12:42:45.655637831 +0100
@@ -283,20 +283,6 @@ TEST_P(TlsConnectStreamPre13, Configured
             ssl_sig_rsa_pss_rsae_sha256);
 }
 
-TEST_P(TlsKeyExchangeTest, Curve25519) {
-  Reset(TlsAgent::kServerEcdsa256);
-  const std::vector<SSLNamedGroup> groups = {
-      ssl_grp_ec_curve25519, ssl_grp_ec_secp256r1, ssl_grp_ec_secp521r1};
-  EnsureKeyShareSetup();
-  ConfigNamedGroups(groups);
-  Connect();
-
-  CheckKeys(ssl_kea_ecdh, ssl_grp_ec_curve25519, ssl_auth_ecdsa,
-            ssl_sig_ecdsa_secp256r1_sha256);
-  const std::vector<SSLNamedGroup> shares = {ssl_grp_ec_curve25519};
-  CheckKEXDetails(groups, shares);
-}
-
 TEST_P(TlsConnectGenericPre13, GroupPreferenceServerPriority) {
   EnsureTlsSetup();
   client_->DisableAllCiphers();
