diff -up ./nss/lib/ssl/sslsock.c.cbcrandomivoff ./nss/lib/ssl/sslsock.c
--- ./nss/lib/ssl/sslsock.c.cbcrandomivoff	2019-08-27 16:37:03.446168087 -0700
+++ ./nss/lib/ssl/sslsock.c	2019-08-27 16:40:31.760827515 -0700
@@ -73,7 +73,7 @@ static sslOptions ssl_defaults = {
     .enableRenegotiation = SSL_RENEGOTIATE_TRANSITIONAL,
     .requireSafeNegotiation = PR_FALSE,
     .enableFalseStart = PR_FALSE,
-    .cbcRandomIV = PR_TRUE,
+    .cbcRandomIV = PR_FALSE,
     .enableOCSPStapling = PR_FALSE,
     .enableALPN = PR_TRUE,
     .reuseServerECDHEKey = PR_TRUE,
@@ -3729,9 +3729,9 @@ ssl_SetDefaultsFromEnvironment(void)
                        PR_TRUE));
         }
         ev = PR_GetEnvSecure("NSS_SSL_CBC_RANDOM_IV");
-        if (ev && ev[0] == '0') {
-            ssl_defaults.cbcRandomIV = PR_FALSE;
-            SSL_TRACE(("SSL: cbcRandomIV set to 0"));
+        if (ev && ev[0] == '1') {
+            ssl_defaults.cbcRandomIV = PR_TRUE;
+            SSL_TRACE(("SSL: cbcRandomIV set to 1"));
         }
     }
 #endif /* NSS_HAVE_GETENV */
