diff -up nss/lib/ssl/sslsock.c.keep_tls_default nss/lib/ssl/sslsock.c
--- nss/lib/ssl/sslsock.c.keep_tls_default	2015-06-05 15:23:25.816895506 -0700
+++ nss/lib/ssl/sslsock.c	2015-06-05 15:24:05.343176138 -0700
@@ -89,13 +89,13 @@ static sslOptions ssl_defaults = {
  * default range of enabled SSL/TLS protocols
  */
 static SSLVersionRange versions_defaults_stream = {
-    SSL_LIBRARY_VERSION_TLS_1_0,
-    SSL_LIBRARY_VERSION_TLS_1_2
+    SSL_LIBRARY_VERSION_3_0,
+    SSL_LIBRARY_VERSION_TLS_1_0
 };
 
 static SSLVersionRange versions_defaults_datagram = {
     SSL_LIBRARY_VERSION_TLS_1_1,
-    SSL_LIBRARY_VERSION_TLS_1_2
+    SSL_LIBRARY_VERSION_TLS_1_1
 };
 
 #define VERSIONS_DEFAULTS(variant) \
