diff -up nss/gtests/ssl_gtest/ssl_skip_unittest.cc.disable-tls13-gtests nss/gtests/ssl_gtest/ssl_skip_unittest.cc
--- nss/gtests/ssl_gtest/ssl_skip_unittest.cc.disable-tls13-gtests	2017-10-16 17:13:51.798825185 +0200
+++ nss/gtests/ssl_gtest/ssl_skip_unittest.cc	2017-10-16 17:14:08.238496409 +0200
@@ -234,6 +234,8 @@ INSTANTIATE_TEST_CASE_P(
 INSTANTIATE_TEST_CASE_P(SkipVariants, TlsSkipTest,
                         ::testing::Combine(TlsConnectTestBase::kTlsVariantsAll,
                                            TlsConnectTestBase::kTlsV11V12));
+#if 0
 INSTANTIATE_TEST_CASE_P(Skip13Variants, Tls13SkipTest,
                         TlsConnectTestBase::kTlsVariantsAll);
+#endif
 }  // namespace nss_test
