diff -up nss/gtests/pk11_gtest/pk11_prf_unittest.cc.disable-ems-gtests nss/gtests/pk11_gtest/pk11_prf_unittest.cc
--- nss/gtests/pk11_gtest/pk11_prf_unittest.cc.disable-ems-gtests	2017-10-30 10:38:09.000000000 +0100
+++ nss/gtests/pk11_gtest/pk11_prf_unittest.cc	2017-11-08 17:35:00.875061831 +0100
@@ -191,37 +191,4 @@ TEST_F(TlsPrfTest, ExtendedMsParamErr) {
   CheckForError(CKM_SHA256, kIncorrectSize, kPmsSize, 0);
 }
 
-// Test matrix:
-//
-//            DH  RSA
-//  TLS_PRF   1   2
-//  SHA256    3   4
-TEST_F(TlsPrfTest, ExtendedMsDhTlsPrf) {
-  Init();
-  ComputeAndVerifyMs(CKM_NSS_TLS_EXTENDED_MASTER_KEY_DERIVE_DH, CKM_TLS_PRF,
-                     nullptr, kExpectedOutputEmsTlsPrf);
-}
-
-TEST_F(TlsPrfTest, ExtendedMsRsaTlsPrf) {
-  Init();
-  ComputeAndVerifyMs(CKM_NSS_TLS_EXTENDED_MASTER_KEY_DERIVE, CKM_TLS_PRF,
-                     &pms_version_, kExpectedOutputEmsTlsPrf);
-  EXPECT_EQ(0, pms_version_.major);
-  EXPECT_EQ(1, pms_version_.minor);
-}
-
-TEST_F(TlsPrfTest, ExtendedMsDhSha256) {
-  Init();
-  ComputeAndVerifyMs(CKM_NSS_TLS_EXTENDED_MASTER_KEY_DERIVE_DH, CKM_SHA256,
-                     nullptr, kExpectedOutputEmsSha256);
-}
-
-TEST_F(TlsPrfTest, ExtendedMsRsaSha256) {
-  Init();
-  ComputeAndVerifyMs(CKM_NSS_TLS_EXTENDED_MASTER_KEY_DERIVE, CKM_SHA256,
-                     &pms_version_, kExpectedOutputEmsSha256);
-  EXPECT_EQ(0, pms_version_.major);
-  EXPECT_EQ(1, pms_version_.minor);
-}
-
 }  // namespace nss_test
diff -up nss/gtests/ssl_gtest/manifest.mn.disable-ems-gtests nss/gtests/ssl_gtest/manifest.mn
--- nss/gtests/ssl_gtest/manifest.mn.disable-ems-gtests	2017-11-08 17:35:00.875061831 +0100
+++ nss/gtests/ssl_gtest/manifest.mn	2017-11-08 17:35:26.531480998 +0100
@@ -21,7 +21,6 @@ CPPSRCS = \
       ssl_dhe_unittest.cc \
       ssl_drop_unittest.cc \
       ssl_ecdh_unittest.cc \
-      ssl_ems_unittest.cc \
       ssl_exporter_unittest.cc \
       ssl_extension_unittest.cc \
       ssl_fragment_unittest.cc \
