diff -up ./cmd/selfserv/selfserv.c.abi_tests ./cmd/selfserv/selfserv.c
--- ./cmd/selfserv/selfserv.c.abi_tests	2019-08-29 14:49:39.183133115 -0700
+++ ./cmd/selfserv/selfserv.c	2019-08-29 14:49:39.216133067 -0700
@@ -393,7 +393,7 @@ printSecurityInfo(PRFileDesc *fd)
                     channel.authKeyBits, suite.authAlgorithmName,
                     channel.keaKeyBits, suite.keaTypeName,
                     channel.compressionMethodName,
-                    channel.extendedMasterSecretUsed ? "Yes" : "No");
+		    channel.reservedNotSupported ? "Yes": "No");
         }
     }
     if (verbose) {
diff -up ./cmd/tstclnt/tstclnt.c.abi_tests ./cmd/tstclnt/tstclnt.c
--- ./cmd/tstclnt/tstclnt.c.abi_tests	2019-08-29 14:49:39.216133067 -0700
+++ ./cmd/tstclnt/tstclnt.c	2019-08-29 14:51:18.683986752 -0700
@@ -177,7 +177,7 @@ printSecurityInfo(PRFileDesc *fd)
                     channel.authKeyBits, suite.authAlgorithmName,
                     channel.keaKeyBits, suite.keaTypeName,
                     channel.compressionMethodName,
-                    channel.extendedMasterSecretUsed ? "Yes" : "No",
+                    channel.reservedNotSupported ? "Yes" : "No",
                     signatureSchemeName(channel.signatureScheme));
         }
     }
diff -up ./gtests/ssl_gtest/tls_agent.cc.abi_tests ./gtests/ssl_gtest/tls_agent.cc
--- ./gtests/ssl_gtest/tls_agent.cc.abi_tests	2019-05-10 14:14:18.000000000 -0700
+++ ./gtests/ssl_gtest/tls_agent.cc	2019-08-29 14:49:39.216133067 -0700
@@ -858,7 +858,7 @@ void TlsAgent::CheckExtendedMasterSecret
   if (version() >= SSL_LIBRARY_VERSION_TLS_1_3) {
     expected = PR_TRUE;
   }
-  ASSERT_EQ(expected, info_.extendedMasterSecretUsed != PR_FALSE)
+  ASSERT_EQ(expected, info_.reservedNotSupported != PR_FALSE)
       << "unexpected extended master secret state for " << name_;
 }
 
