diff -up nss/tests/cert/cert.sh.skip-rsa-exponent-tests nss/tests/cert/cert.sh
--- nss/tests/cert/cert.sh.skip-rsa-exponent-tests	2017-11-09 16:11:24.558501261 +0100
+++ nss/tests/cert/cert.sh	2017-11-09 16:11:33.944277815 +0100
@@ -1337,10 +1337,6 @@ MODSCRIPT
   CU_ACTION="Setting invalid database password in FIPS mode"
   RETEXPECTED=255
   certu -W -d "${PROFILEDIR}" -f "${R_FIPSPWFILE}" -@ "${R_FIPSBADPWFILE}" 2>&1
-  CU_ACTION="Attempt to generate a key with exponent of 3 (too small)"
-  certu -G -k rsa -g 2048 -y 3 -d "${PROFILEDIR}" -z ${R_NOISE_FILE} -f "${R_FIPSPWFILE}" 
-  CU_ACTION="Attempt to generate a key with exponent of 17 (too small)"
-  certu -G -k rsa -g 2048 -y 17 -d "${PROFILEDIR}" -z ${R_NOISE_FILE} -f "${R_FIPSPWFILE}" 
   RETEXPECTED=0
 
   CU_ACTION="Generate Certificate for ${CERTNAME}"
