diff -up nss/cmd/selfserv/selfserv.c.abi_tests nss/cmd/selfserv/selfserv.c
--- nss/cmd/selfserv/selfserv.c.abi_tests	2016-08-16 12:36:23.695996680 +0200
+++ nss/cmd/selfserv/selfserv.c	2016-08-16 12:39:00.006879649 +0200
@@ -425,7 +425,7 @@ printSecurityInfo(PRFileDesc *fd)
                     channel.authKeyBits, suite.authAlgorithmName,
                     channel.keaKeyBits, suite.keaTypeName,
                     channel.compressionMethodName,
-                    channel.extendedMasterSecretUsed ? "Yes" : "No");
+		    channel.reservedNotSupported ? "Yes": "No");
         }
     }
     if (verbose) {
diff -up nss/cmd/tstclnt/tstclnt.c.abi_tests nss/cmd/tstclnt/tstclnt.c
--- nss/cmd/tstclnt/tstclnt.c.abi_tests	2016-08-16 12:36:23.696996653 +0200
+++ nss/cmd/tstclnt/tstclnt.c	2016-08-16 12:39:24.460235581 +0200
@@ -129,7 +129,7 @@ printSecurityInfo(PRFileDesc *fd)
                     channel.authKeyBits, suite.authAlgorithmName,
                     channel.keaKeyBits, suite.keaTypeName,
                     channel.compressionMethodName,
-                    channel.extendedMasterSecretUsed ? "Yes" : "No");
+		    channel.reservedNotSupported ? "Yes": "No");
         }
     }
     cert = SSL_RevealCert(fd);
diff -up nss/gtests/ssl_gtest/tls_agent.cc.abi_tests nss/gtests/ssl_gtest/tls_agent.cc
--- nss/gtests/ssl_gtest/tls_agent.cc.abi_tests	2016-08-16 12:36:23.696996653 +0200
+++ nss/gtests/ssl_gtest/tls_agent.cc	2016-08-16 12:39:45.167690174 +0200
@@ -571,7 +571,7 @@ void TlsAgent::CheckExtendedMasterSecret
   if (version() >= SSL_LIBRARY_VERSION_TLS_1_3) {
     expected = PR_TRUE;
   }
-  ASSERT_EQ(expected, info_.extendedMasterSecretUsed != PR_FALSE)
+  ASSERT_EQ(expected, info_.reservedNotSupported != PR_FALSE)
       << "unexpected extended master secret state for " << name_;
 }
 
