diff -up nss/gtests/manifest.mn.skip_util_gtest nss/gtests/manifest.mn
--- nss/gtests/manifest.mn.skip_util_gtest	2017-11-08 17:22:45.833700300 +0100
+++ nss/gtests/manifest.mn	2017-11-08 17:24:01.320991577 +0100
@@ -33,6 +33,5 @@ endif
 
 DIRS = \
 	$(LIB_SRCDIRS) \
-	$(UTIL_SRCDIRS) \
 	$(NSS_SRCDIRS) \
 	$(NULL)
diff -up nss/gtests/ssl_gtest/manifest.mn.skip_util_gtest nss/gtests/ssl_gtest/manifest.mn
