diff -up nss/cmd/Makefile.skipthem nss/cmd/Makefile
--- nss/cmd/Makefile.skipthem	2017-01-13 16:41:04.117486801 +0100
+++ nss/cmd/Makefile	2017-01-13 16:42:31.396335957 +0100
@@ -19,7 +19,11 @@ BLTEST_SRCDIR =
 ECPERF_SRCDIR =
 FREEBL_ECTEST_SRCDIR =
 FIPSTEST_SRCDIR =
+ifeq ($(NSS_BLTEST_NOT_AVAILABLE),1)
+SHLIBSIGN_SRCDIR = shlibsign
+else
 SHLIBSIGN_SRCDIR =
+endif
 else
 BLTEST_SRCDIR = bltest
 ECPERF_SRCDIR = ecperf
